### PR TITLE
Fix some -Wfloat-conversion warnings

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: src/core/screen.c [hunk 1]
```diff
@@ -1239,14 +1240,14 @@ get_window_surface (MetaWindow *window)
   /* Scale surface to fit current screen */
   if (width > height)
     {
-      max_size = MIN (MAX_PREVIEW_SIZE, current->rect.width / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
+      max_size = MIN (MAX_PREVIEW_SIZE, MAX_PREVIEW_SCREEN_FRACTION * ((double) current->rect.width) / ((double) max_columns));
```
Comment (raveit65) [inline src/core/screen.c:1243]: Should it not be ?
```
- max_size = MIN (MAX_PREVIEW_SIZE, current->rect.width / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
+ max_size = MIN (MAX_PREVIEW_SIZE, ((double) current->rect.width) /  MAX_PREVIEW_SCREEN_FRACTION * ((double) max_columns));
```

File: src/core/screen.c [hunk 2]
```diff
@@ -1239,14 +1240,14 @@ get_window_surface (MetaWindow *window)
...
       ratio = ((double) width) / max_size;
       width = (int) max_size;
       height = (int) (((double) height) / ratio);
     }
   else
     {
-      max_size = MIN (MAX_PREVIEW_SIZE, current->rect.height / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
+      max_size = MIN (MAX_PREVIEW_SIZE, MAX_PREVIEW_SCREEN_FRACTION * ((double) current->rect.height) / ((double) max_columns));
```
Comment (raveit65) [inline src/core/screen.c:1250]: Same here?

File: src/core/screen.c (hunk 1 above)
Comment (raveit65) [reply src/core/screen.c:1243]: Not sure if subtraction part is done before multiplication here in C.

Comment (rbuj) [reply src/core/screen.c:1243]:
```
$ cat << EOF > polish_notation.c
#include <stdio.h>

int
main (void)
{
  printf ("%d\n", 8 / 4 * 2);
  printf ("%d\n", (8 / 4) * 2);
  return 0;
}
EOF
gcc -Wall polish_notation.c -o polish_notation
./polish_notation 
```
Result:
```
4
4
```

Comment (rbuj) [reply src/core/screen.c:1243]: The operators '*' and '/' have the same preference, so they operations are made from left to right.

Comment (rbuj) [reply src/core/screen.c:1243]: where is the subtraction?

Comment (raveit65) [reply src/core/screen.c:1243]: Opps, my translating mistake. I meant division.

> The operators '*' and '/' have the same preference, so the operations are made from left to right.

Thanks for explanation.

**Review by raveit65 — APPROVED** [review]
LGTM

**Comment by raveit65** [comment]
@mate-desktop/core-team 
Any one like to review this one?

**Pull request merged**